src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Primitives/Popup.cs

```diff
@@ -3571,6 +3571,13 @@ internal static bool IsPerMonitorDpiScalingActive
                         return new NativeMethods.POINT((int)screenOrigin.X, (int)screenOrigin.Y);
                     }
                 }
+                else
+                {
+                    // GetPlacementTargetInterestPoints already returns points in screen-coordinate
+                    Point[] placementTargetInterestPoints = popup.GetPlacementTargetInterestPoints(popup.PlacementInternal);
+                    Point center = placementTargetInterestPoints[(int)InterestPoint.TopLeft];
```

Comment:
Thank you, can we be sure that there will be no array IndexOutOfRange problem?

Reply:
Yes. It is created by this method:

https://github.com/dotnet/wpf/blob/b9b48871d457fc1f78fa9526c0570dae8e34b488/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Primitives/Popup.cs#L2415-L2428

And consumed similarly elsewhere, see e.g. `GetPopupRootLimits` or `UpdatePosition` methods in the same file.